### PR TITLE
2nd order Taylor expansions

### PR DESCRIPTION
**Pull request by Giannis1993**
Corrected the `SphericalTaylor2` and `NonSphericalTaylor2` classes.

### CODE REVIEW

File: examples/polynomial1D.py [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
-from sao.move_limits import Bounds, MoveLimit
+from sao.approximations import Taylor1, Taylor2, SphericalTaylor2, NonSphericalTaylor2
+from sao.problems.subproblem import Subproblem
+from sao.intervening_variables.exponential import Linear
+from sao.intervening_variables.mma import MMA02
+from sao.intervening_variables.mixed_intervening import MixedIntervening
+from sao.move_limits import Bounds, MoveLimit, MoveLimitFraction, AdaptiveMoveLimit
 from sao.util import Plot
-from sao.solvers import SvanbergIP
+from sao.solvers import SolverIP_Svanberg
```
Comment (artofscience) [inline examples/polynomial1D.py:10]: you do not have to import those, not using them

File: examples/polynomial1D.py [hunk 2]
```diff
@@ -25,52 +28,39 @@ def example_polynomial_1D():
...
-    # criterion = KKT(x_min=prob.x_min, x_max=prob.x_max)
-    # criterion = ObjectiveChange()
-    # criterion = VariableChange(x_min=prob.x_min, x_max=prob.x_max)
-    # criterion = Feasibility()
-    # criterion = Alltogether(x_min=prob.x_min, x_max=prob.x_max)
+    problem = Polynomial1D()
+    bounds = Bounds(xmin=problem.x_min, xmax=problem.x_max)
+    movelimit = MoveLimitFraction(fraction=2)
+    intvar = Linear()     # MMA02(x_min=problem.x_min, x_max=problem.x_max)
+    subproblem = Subproblem(SphericalTaylor2(intvar), limits=[bounds, movelimit])
```
Comment (artofscience) [inline examples/polynomial1D.py:35]: equates to SphericalTaylor2()

File: examples/polynomial1D.py (hunk 1 above)
Comment (artofscience) [inline examples/polynomial1D.py:10]:
```suggestion

```

File: examples/polynomial1D.py (hunk 2 above)
Comment (artofscience) [inline examples/polynomial1D.py:35]:
```suggestion
    subproblem = Subproblem(SphericalTaylor2(), limits=[bounds, movelimit])
```

Comment (Giannis1993) [reply examples/polynomial1D.py:35]: I wanted to have it clearly written so i could change the `intvar` around.

File: examples/polynomial1D.py [hunk 3]
```diff
@@ -150,5 +124,5 @@ def example_polynomial_1D_mixed():
 
 
 if __name__ == "__main__":
-    example_polynomial_1D()
+    # example_polynomial_1D()
```
Comment (artofscience) [inline examples/polynomial1D.py:127]:
```suggestion
    example_polynomial_1D()
```

**Comment by artofscience** [comment]
@Giannis1993 can you maybe come up with some tests? e.g. you can (after making the approximation), test whether function values/gradient are fitting?

**Review by artofscience — APPROVED** [review]

**Pull request merged**